### DISCUSSION TITLE
commandes MQTT pour actions

### DISCUSSION THREAD

**Discussion by pes69** [echavet/MitsubishiCN105ESPHome#155] [Q&A]
Salut@echavet,
J'ai ajouté la prise charge pour avoir la gestion en MQTT dans Jeedom, tout est ok mais je n'ai pas les commandes actions, juste les infos à la difference d'avec Mitsubishi2MQTT. Y a t'il quelque chose à rajouter dans le fichier yaml ?

**Comment by echavet** [comment]
Salut @pes69 
non tu n'as rien à rajouter mis à part bien entendu les paramètres de ton broker mqtt.
peux-tu fournir ton yaml complet stp?
Il est possible qu'il y ait un bug avec le support esp-idf de mqtt (d'après la doc).
Sinon il faut bien vérifier que les topics par défaut correspondent à ceux attendus par ton serveur jeedom.
Mon intégration MQTT retrouve bien toutes les actions avec Home Assistant. As-tu un moyen de voir quels sont les messages de config reçus par jeedom ?

**Comment by pes69** [comment]
Merci pour ta réponse.
Ci-joint le fichier .YAML modifié.
[hvac_esp32_rdc - Copie.yaml.TXT](https://github.com/user-attachments/files/16818797/hvac_esp32_rdc.-.Copie.yaml.TXT)
Pour info, je n'ai pas pu encore plugger mon ESP32-V3 à mes clims car j'ai du commander 2 connecteurs 4 pins que je n'avais pas; peut-être que c'est pour cela que rien ne remonte dans Jeedom ?
J'ai aussi utilisé MQTT Explorer pour voir tout ce qui était remonté dans mes Topic, uniquement que les infos, pas les commandes actions.
En tout cas un grand merci pour ton travail.

**Reply by echavet** [reply]
Merci.

Tout semble correct.
A ta place, je testerais en définissant manuellement les topics d'action. Tu peux essayer de reproduire ceux utilisés sans Mitsubishi2MQTT.
Bon courage. Tiens nous au courant si tu t'en sors.

**Comment by pes69** [comment]
Je peux récupérer les commandes MQTT action dans le fichier livré avec Mitsubishi2MQTT que j'ai injecté avec Arduino ?
Tu pourrais m'en donner une STP pour exemple ?

**Comment by echavet** [comment]
et bien ils sont dans ce fichier:
https://github.com/gysmo38/mitsubishi2MQTT/blob/master/src/mitsubishi2mqtt/mitsubishi2mqtt.ino#L159
sinon pour le format exact, tu essayer d'utiliser un outil comme MQTT.fx (ou https://github.com/thomasnordquist/MQTT-Explorer que je ne connais pas). Tu écoutes le topic homeassistant et tu regardes ce qu'envoie Mitsubishi2MQTT et tu compares avec ce qu'envoie ESPHome.
Je sais que c'est un travail laborieux mais je n'ai pas la moindre idée de la raison pour laquelle jeedom ne traite pas les payload de config mqtt. 
Et avec espHome tout est automatique. Je n'ai jamais développé quoi que ce soit pour le support MQTT. Je peux juste te confirmer qu'avec Home Assistant, ça fonctionne. Ce qui signifie que les trames de configuration sont envoyées par le firmware.

**Comment by pes69** [comment]
Salut@echavet,
Retour de vacances, connexion du ATOM S3 Lite à ma clim MSZ-AP35VGK, l'ATOM S3 Lite est bien alimenté, je le retrouve bien dans ESP Home, mais rien ne remonte à part cela dans les LOGS :
[01:05:51][I][CN105:008]: setupUART() with baudrate  2400 
[01:05:51][W][CN105:162]: Heatpump has not replied for 533 s
[01:05:51][I][CN105:163]: We think Heatpump is not connected anymore..

J'ai inversé les Rx et Tx, rien, toujours rien.
Si tu as une idée STP.

**Comment by pes69** [comment]
Salut@echavet,,
Je viens de refaire la connectique, c'est good, je vois la Clim, encore merci pour ton aide.
Bonne soirée.

**Comment by pes69** [comment]
Salut@echavet,
Impossible de trouver les commandes action MQTT hormis pour la puissance Fan (low, middle, ...).
J'ai tout essayé mais pas moyen de récupérer les autres commandes action.
Etant sous Jeedom / MQTT, je n'ai aucun autre moyen de commander la clim.
Les commandes Mitsubishi2MQTT ne fonctionnent pas (d'ailleurs la gestion du Fan n'est pas la même).
Aurais-tu une suggestion STP ?
PES

**Reply by echavet** [reply]
Non je sais pas trop.
A part sniffer les paquets MQTT avec un client mqtt dans un contexte fonctionnel. Par exemple avec Mitsubishi2MQTT. Tu m'avais pas dit que ça fonctionnait bien avec Mitsubishi2MQTT et jeedom?

**Comment by pes69** [comment]
Oui, avec Mitsubishi2MQTT cela fonctionne, j'ai 2 splits actuellement, j'en ai 1 géré avec Mitsubishi2MQTT et 1 autre avec MitsubishiCN105ESPHOME, j'ai fait une capture de tout ce qui est remonté sur les 2 topics avec MQTTExplorer, tu peux voir que les commandes sont bien différentes :
![image](https://github.com/user-attachments/assets/84b742b4-2e2f-4b2b-b4db-12cbf5f9011e)

**Comment by pes69** [comment]
je viens d'allumer la clim avec la télécommande de l'ESP32, j'ai fait une capture de tout ce qui est en relation avec le Topic HVAC-ESP32-RDC, je ne vois aucune action, que des infos hormis fan_mode et swing_mode où là je peux interagir depuis JEEDOM/MQTT : 
![image](https://github.com/user-attachments/assets/44bc359c-a68a-4ef4-af0a-d72e8442e64a)
